Wrappers/Python/test/test_approximate_gradient.py

```diff
@@ -0,0 +1,238 @@
+#  Copyright 2024 United Kingdom Research and Innovation
+#  Copyright 2024 The University of Manchester
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+# Authors:
+# CIL Developers, listed at: https://github.com/TomographicImaging/CIL/blob/master/NOTICE.txt
+
+
+import unittest
+from utils import initialise_tests
+
+import numpy as np
+
+from cil.framework import VectorData
+
+
+from cil.utilities import dataexample
+from cil.optimisation.functions import LeastSquares
+from cil.optimisation.functions import ApproximateGradientSumFunction
+from cil.optimisation.functions import SGFunction
+from cil.optimisation.operators import MatrixOperator
+from cil.optimisation.algorithms import GD
+from cil.framework import VectorData
+from cil.optimisation.utilities import Sampler, SamplerRandom
+
+from testclass import CCPiTestClass
+from utils import has_astra
+
+initialise_tests()
+
+if has_astra:
+    from cil.plugins.astra import ProjectionOperator
+
+
+class TestApproximateGradientSumFunction(CCPiTestClass):
+
+    def setUp(self):
+        self.sampler = Sampler.random_with_replacement(5)
+        self.initial = VectorData(np.zeros(10))
+        self.b = VectorData(np.random.normal(0, 1, 10))
+        self.functions = []
+        for i in range(5):
+            diagonal = np.zeros(10)
+            diagonal[2*i:2*(i+1)] = 1
+            A = MatrixOperator(np.diag(diagonal))
+            self.functions.append(LeastSquares(A, A.direct(self.b)))
+            if i == 0:
+                self.objective = LeastSquares(A, A.direct(self.b))
+            else:
+                self.objective += LeastSquares(A, A.direct(self.b))
+
+    def test_ABC(self):
+        with self.assertRaises(TypeError):
+            self.stochastic_objective = ApproximateGradientSumFunction(
+                self.functions, self.sampler)
+
+
+class TestSGD(CCPiTestClass):
```

Comment:
You can add some [cvxpy tests](https://github.com/epapoutsellis/StochasticCIL/blob/svrg/Wrappers/Python/test/test_SGFunction.py#L98) for the SGD case. I used only the overparameterized case where variance is 0. I think I have the same for the other estimators, all motivated by Figure 3 from https://richtarik.org/papers/sigma_k.pdf

Reply:
Added in a cvxpy test :)